backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -116,20 +125,30 @@ private boolean isExpired(Participant participant) {
   }
 
   /** Send JSON response for expired participant that has been removed from datastore */
-  private void sendExpiredResponse(HttpServletResponse response) throws IOException {
+  private void sendExpiredResponse(HttpServletResponse response, Participant participant)
+      throws IOException {
     JSONObject expired = new JSONObject();
     expired.put(JSON_MATCH_STATUS, "expired");
-
+    try {
+      EMAIL_NOTIFIER.sendExpiredEmail(participant.getUsername());
+    } catch (MessagingException e) {
+      e.printStackTrace();
+    }
     // Send the JSON back as the response
     response.setContentType("application/json");
     response.getWriter().println(expired.toString());
   }
 
   /** Send JSON response for no match yet */
-  private void sendNoMatchResponse(HttpServletResponse response) throws IOException {
+  private void sendNoMatchResponse(HttpServletResponse response, Participant participant)
+      throws IOException {
     JSONObject noMatchYet = new JSONObject();
     noMatchYet.put(JSON_MATCH_STATUS, "false");
-
+    try {
+      EMAIL_NOTIFIER.sendNoMatchEmail(participant.getUsername());
+    } catch (MessagingException e) {
+      e.printStackTrace();
+    }
```

Comment:
No email necessary here!

Reply:
Although it's not  necessary, what are we doing in the case of no match ?

Reply:
Commen above explained